lib/active_merchant/billing/gateways/nuvei.rb

```diff
@@ -162,10 +162,10 @@ def add_account_funding_transaction(post, payment, options = {})
         address_details = {
           firstName: payment.first_name,
           lastName: payment.last_name,
-          country: options.dig(:billing_address, :country),
-          address: options.dig(:billing_address, :address1),
-          city: options.dig(:billing_address, :city),
-          state: options.dig(:billing_address, :state)
+          country: payment.try(:country) || options.dig(:billing_address, :country),
+          address: payment.try(:address1) || payment.try(:address2) || options.dig(:billing_address, :address1),
+          city: payment.try(:city) || options.dig(:billing_address, :city),
+          state: payment.try(:state) || options.dig(:billing_address, :state)
```

Comment:
I don't think this will work since address information is not a part of any of AM's classes related to payment methods.

Reply:
Hi @naashton I thought that too, but I think if they create a payment method with spreedly API, we can get the billing informattion from there ,can't we ?

Reply:
Oh nevermind I just saw your comment on the ticket and it is already covered, thanks